variable_set.go

```diff
@@ -0,0 +1,255 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ VariableSets = (*variableSets)(nil)
+
+// VariableSets describes all the Variable Set related methods that the
+// Terraform Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/cloud-docs/api-docs/variable-sets
+type VariableSets interface {
+	// List all the variable sets within an organization.
+	List(ctx context.Context, organization string, options VariableSetListOptions) (*VariableSetList, error)
+
+	// Create is used to create a new variable set.
+	Create(ctx context.Context, organization string, options VariableSetCreateOptions) (*VariableSet, error)
+
+	// Read a variable set by its ID.
+	Read(ctx context.Context, variableSetID string) (*VariableSet, error)
+
+	// Update an existing variable set.
+	Update(ctx context.Context, variableSetID string, options VariableSetUpdateOptions) (*VariableSet, error)
+
+	// Delete a variable set by ID.
+	Delete(ctx context.Context, variableSetID string) error
+
+	// Assign a variable set to workspaces
+	Assign(ctx context.Context, variableSetID string, options VariableSetAssignOptions) (*VariableSet, error)
+}
+
+type variableSets struct {
+	client *Client
+}
+
+type VariableSetList struct {
+	*Pagination
+	Items []*VariableSet
+}
+
+type VariableSet struct {
+	ID          string `jsonapi:"primary,varsets"`
+	Name        string `jsonapi:"attr,name"`
+	Description string `jsonapi:"attr,description"`
+	Global      bool   `jsonapi:"attr,global"`
+
+	// Relations
+	Workspaces []*Workspace           `jsonapi:"relation,workspaces,omitempty"`
+	Variables  []*VariableSetVariable `jsonapi:"relation,vars,omitempty"`
+}
+
+type VariableSetListOptions struct {
+	ListOptions
+}
+
+func (o VariableSetListOptions) valid() error {
+	return nil
+}
+
+// List all Variable Sets in the organization
+func (s *variableSets) List(ctx context.Context, organization string, options VariableSetListOptions) (*VariableSetList, error) {
```

Comment:
In the [1.0 release](https://github.com/hashicorp/go-tfe/pull/291), all list options will be expressed as a pointer to simplify callers who provide default list options. Maybe we can get ahead of that now?